### ISSUE TITLE
why the loss keep increasing during iteration

### ISSUE THREAD

**Issue by bevanking** [vikasrtr/pyRidge#1]
I output the loss (corresponding to the variable J_all) in each iteration when running RidgeGradientDescent.py. I find that the loss keeps increasing other than decreasing. But the predict results seems make sense. I don't know why.

**Comment by echoKangYL** [comment]
I have the same question,  and cannot find the reason......

**Comment by echoKangYL** [comment]
> I output the loss (corresponding to the variable J_all) in each iteration when running RidgeGradientDescent.py. I find that the loss keeps increasing other than decreasing. But the predict results seems make sense. I don't know why.

I met the same problem and it is really confusing, have you solved it?
Thank you~

**Comment by echoKangYL** [comment]
> I output the loss (corresponding to the variable J_all) in each iteration when running RidgeGradientDescent.py. I find that the loss keeps increasing other than decreasing. But the predict results seems make sense. I don't know why.

I think I get it.
In the function '_compute_cost', the shape of 'X', 'w', 'y' is _(n, d), (d, 1)_ and _(n, )_, respectively, where _'n'_ is the size of the training set and 'd' is the number of features. So, _np.dot(X, w)_ is a _(n, 1)_ vector.
When applying _np.dot(X, w) - y_, there is a _(n, n)_ matrix, rather than a _(n, 1)_ vector.
Then, by _np.sum((np.dot(X, w) - y) ** 2.)_, it will be a large value.
To solve the problem, reshape 'y' is necessary.
Just add _'y = y.reshape(-1, 1)'_ before calculating 'J'.
